services/offline/src/lib/cacheable-section.tsx

```diff
@@ -1,15 +1,33 @@
 import PropTypes from 'prop-types'
 import React, { useEffect } from 'react'
+import { RecordingState } from '../types'
 import { useRecordingState, useCachedSection } from './cacheable-section-state'
 import { useOfflineInterface } from './offline-interface'
 
-const recordingStates = {
+const recordingStates: { [index: string]: RecordingState } = {
```

Comment:
This map can be replaced with [an enum](https://www.typescriptlang.org/docs/handbook/enums.html):

```typescript
enum RecordingState {
    Default,
    Pending,
    Recording,
    Error
}
```

which can then be used like so: `setRecordingState(RecordingState.Error)`

Reply:
Cool 👍 good to know for the future though!